addons/medical_status/functions/fnc_setDead.sqf

```diff
@@ -22,8 +23,17 @@ _unit setVariable [VAR_BLOOD_PRESS, [0, 0], true];
 
 _unit setVariable [QEGVAR(medical,causeOfDeath), _reason, true];
 
+// Get last instigator in case this isn't called manually
+if (isNull _instigator) then {
+    _instigator = _unit getVariable [QEGVAR(medical,lastInstigator), objNull];
+};
+
 // Send a local event before death
 [QEGVAR(medical,death), [_unit]] call CBA_fnc_localEvent;
 
 // Kill the unit without changing visual apperance
-[_unit, 1] call EFUNC(medical_engine,setStructuralDamage);
+private _prevDamage = _unit getHitPointDamage "hitHead";
+
+_unit setHitPointDamage ["hitHead", 1, true, _instigator];
+
+_unit setHitPointDamage ["hitHead", _prevDamage, true, _instigator];
```

Comment:
I'm not sure if the instigator parameter makes semantic sense for the second call here to reset the damage.

Unlikely to have meaningful effect, but for the sake of clear code intentions I'd personally leave it off.

Reply:
I'll just drop the latter two parameters in the second call.